### ISSUE TITLE
nghttpx: error while loading shared libraries: libnghttp2.so

### ISSUE THREAD

**Issue by Tapanito** [nghttp2/nghttp2#916]
I cloned the master repository and followed installation instructions provided in the documentation. I installed the nghttp2 suite using make install, however when I try to run nghttpx I get the following error:

```
nghttpx: error while loading shared libraries: libnghttp2.so.14: cannot open shared object file: No such file or directory
```
It seems that make install does not copy the shared libraries to the system.

Running nghttpx from the <repo>/src directory works fine.

**Comment by tatsuhiro-t** [comment]
Have you run `sudo ldconfig -v`?

**Comment by Tapanito** [comment]
That did the trick. 
Would it be possible to include that in the make file?

**Comment by tatsuhiro-t** [comment]
I think that the instruction to run ldconfig is displayed when `make install` is executed.

**Comment by Tapanito** [comment]
This is a copy paste from running `make install` 
I can't find the instruction to run the command.
```
Making install in lib
make[1]: Entering directory '/nghttp2-master/lib'
Making install in includes
make[2]: Entering directory '/nghttp2-master/lib/includes'
make[3]: Entering directory '/nghttp2-master/lib/includes'
make[3]: Nothing to be done for 'install-exec-am'.
 /bin/mkdir -p '/usr/local/include'
 /bin/mkdir -p '/usr/local/include/nghttp2'
 /usr/bin/install -c -m 644  nghttp2/nghttp2.h nghttp2/nghttp2ver.h '/usr/local/include/nghttp2'
make[3]: Leaving directory '/nghttp2-master/lib/includes'
make[2]: Leaving directory '/nghttp2-master/lib/includes'
make[2]: Entering directory '/nghttp2-master/lib'
  CC       nghttp2_pq.lo
  CC       nghttp2_map.lo
  CC       nghttp2_queue.lo
  CC       nghttp2_frame.lo
  CC       nghttp2_buf.lo
  CC       nghttp2_stream.lo
  CC       nghttp2_outbound_item.lo
  CC       nghttp2_session.lo
  CC       nghttp2_submit.lo
  CC       nghttp2_helper.lo
  CC       nghttp2_npn.lo
  CC       nghttp2_hd.lo
  CC       nghttp2_hd_huffman.lo
  CC       nghttp2_hd_huffman_data.lo
  CC       nghttp2_version.lo
  CC       nghttp2_priority_spec.lo
  CC       nghttp2_option.lo
  CC       nghttp2_callbacks.lo
  CC       nghttp2_mem.lo
  CC       nghttp2_http.lo
  CC       nghttp2_rcbuf.lo
  CC       nghttp2_debug.lo
  CCLD     libnghttp2.la
make[3]: Entering directory '/nghttp2-master/lib'
 /bin/mkdir -p '/usr/local/lib'
 /bin/bash ../libtool   --mode=install /usr/bin/install -c   libnghttp2.la '/usr/local/lib'
libtool: install: /usr/bin/install -c .libs/libnghttp2.so.14.13.2 /usr/local/lib/libnghttp2.so.14.13.2
libtool: install: (cd /usr/local/lib && { ln -s -f libnghttp2.so.14.13.2 libnghttp2.so.14 || { rm -f libnghttp2.so.14 && ln -s libnghttp2.so.14.13.2 libnghttp2.so.14; }; })
libtool: install: (cd /usr/local/lib && { ln -s -f libnghttp2.so.14.13.2 libnghttp2.so || { rm -f libnghttp2.so && ln -s libnghttp2.so.14.13.2 libnghttp2.so; }; })
libtool: install: /usr/bin/install -c .libs/libnghttp2.lai /usr/local/lib/libnghttp2.la
libtool: install: /usr/bin/install -c .libs/libnghttp2.a /usr/local/lib/libnghttp2.a
libtool: install: chmod 644 /usr/local/lib/libnghttp2.a
libtool: install: ranlib /usr/local/lib/libnghttp2.a
libtool: finish: PATH="/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/sbin" ldconfig -n /usr/local/lib
----------------------------------------------------------------------
Libraries have been installed in:
   /usr/local/lib

If you ever happen to want to link against installed libraries
in a given directory, LIBDIR, you must either use libtool, and
specify the full pathname of the library, or use the `-LLIBDIR'
flag during linking and do at least one of the following:
   - add LIBDIR to the `LD_LIBRARY_PATH' environment variable
     during execution
   - add LIBDIR to the `LD_RUN_PATH' environment variable
     during linking
   - use the `-Wl,-rpath -Wl,LIBDIR' linker flag
   - have your system administrator add LIBDIR to `/etc/ld.so.conf'

See any operating system documentation about shared libraries for
more information, such as the ld(1) and ld.so(8) manual pages.
----------------------------------------------------------------------
 /bin/mkdir -p '/usr/local/lib/pkgconfig'
 /usr/bin/install -c -m 644 libnghttp2.pc '/usr/local/lib/pkgconfig'
make[3]: Leaving directory '/nghttp2-master/lib'
make[2]: Leaving directory '/nghttp2-master/lib'
make[1]: Leaving directory '/nghttp2-master/lib'
Making install in third-party
make[1]: Entering directory '/nghttp2-master/third-party'
  CC       http-parser/http_parser.lo
  CCLD     libhttp-parser.la
make[2]: Entering directory '/nghttp2-master/third-party'
make[2]: Nothing to be done for 'install-exec-am'.
make[2]: Nothing to be done for 'install-data-am'.
make[2]: Leaving directory '/nghttp2-master/third-party'
make[1]: Leaving directory '/nghttp2-master/third-party'
Making install in src
make[1]: Entering directory '/nghttp2-master/src'
Making install in includes
make[2]: Entering directory '/nghttp2-master/src/includes'
make[3]: Entering directory '/nghttp2-master/src/includes'
make[3]: Nothing to be done for 'install-exec-am'.
make[3]: Leaving directory '/nghttp2-master/src/includes'
make[2]: Leaving directory '/nghttp2-master/src/includes'
make[2]: Entering directory '/nghttp2-master/src'
  CXX      libnghttpx_a-util.o
  CXX      libnghttpx_a-http2.o
  CC       libnghttpx_a-timegm.o
  CXX      libnghttpx_a-app_helper.o
x  CXX      libnghttpx_a-tls.o
  CXX      libnghttpx_a-shrpx_config.o
  CXX      libnghttpx_a-shrpx_accept_handler.o
  CXX      libnghttpx_a-shrpx_connection_handler.o
  CXX      libnghttpx_a-shrpx_client_handler.o
  CXX      libnghttpx_a-shrpx_http2_upstream.o
  CXX      libnghttpx_a-shrpx_https_upstream.o
  CXX      libnghttpx_a-shrpx_downstream.o
  CXX      libnghttpx_a-shrpx_downstream_connection.o
  CXX      libnghttpx_a-shrpx_http_downstream_connection.o
  CXX      libnghttpx_a-shrpx_http2_downstream_connection.o
  CXX      libnghttpx_a-shrpx_http2_session.o
  CXX      libnghttpx_a-shrpx_downstream_queue.o
  CXX      libnghttpx_a-shrpx_log.o
  CXX      libnghttpx_a-shrpx_http.o
  CXX      libnghttpx_a-shrpx_io_control.o
  CXX      libnghttpx_a-shrpx_tls.o
  CXX      libnghttpx_a-shrpx_worker.o
  CXX      libnghttpx_a-shrpx_log_config.o
  CXX      libnghttpx_a-shrpx_connect_blocker.o
  CXX      libnghttpx_a-shrpx_live_check.o
  CXX      libnghttpx_a-shrpx_downstream_connection_pool.o
  CXX      libnghttpx_a-shrpx_rate_limit.o
  CXX      libnghttpx_a-shrpx_connection.o
  CXX      libnghttpx_a-shrpx_memcached_dispatcher.o
  CXX      libnghttpx_a-shrpx_memcached_connection.o
  CXX      libnghttpx_a-shrpx_worker_process.o
  CXX      libnghttpx_a-shrpx_signal.o
  CXX      libnghttpx_a-shrpx_router.o
  CXX      libnghttpx_a-shrpx_api_downstream_connection.o
  CXX      libnghttpx_a-shrpx_health_monitor_downstream_connection.o
  CXX      libnghttpx_a-shrpx_exec.o
  CXX      libnghttpx_a-shrpx_dns_resolver.o
  CXX      libnghttpx_a-shrpx_dual_dns_resolver.o
  CXX      libnghttpx_a-shrpx_dns_tracker.o
  CC       libnghttpx_a-xsi_strerror.o
  AR       libnghttpx.a
  CXX      util.o
  CXX      http2.o
  CC       timegm.o
  CXX      app_helper.o
  CC       nghttp2_gzip.o
  CXX      nghttp.o
  CXX      HtmlParser.o
  CXX      tls.o
  CXXLD    nghttp
  CXX      nghttpd.o
  CXX      HttpServer.o
  CXXLD    nghttpd
  CXX      nghttpx-shrpx.o
  CXXLD    nghttpx
  CXX      h2load.o
  CXX      h2load_http2_session.o
  CXX      h2load_http1_session.o
  CXXLD    h2load
  CXX      inflatehd.o
  CC       comp_helper.o
  CXXLD    inflatehd
  CXX      deflatehd.o
  CXXLD    deflatehd
make[3]: Entering directory '/nghttp2-master/src'
 /bin/mkdir -p '/usr/local/bin'
  /bin/bash ../libtool   --mode=install /usr/bin/install -c nghttp nghttpd nghttpx h2load inflatehd deflatehd '/usr/local/bin'
libtool: install: /usr/bin/install -c .libs/nghttp /usr/local/bin/nghttp
libtool: install: /usr/bin/install -c .libs/nghttpd /usr/local/bin/nghttpd
libtool: install: /usr/bin/install -c .libs/nghttpx /usr/local/bin/nghttpx
libtool: install: /usr/bin/install -c .libs/h2load /usr/local/bin/h2load
libtool: install: /usr/bin/install -c .libs/inflatehd /usr/local/bin/inflatehd
libtool: install: /usr/bin/install -c .libs/deflatehd /usr/local/bin/deflatehd
make[3]: Leaving directory '/nghttp2-master/src'
make[2]: Leaving directory '/nghttp2-master/src'
make[1]: Leaving directory '/nghttp2-master/src'
Making install in examples
make[1]: Entering directory '/nghttp2-master/examples'
  CC       client.o
  CCLD     client
  CC       libevent-client.o
  CCLD     libevent-client
  CC       libevent-server.o
  CCLD     libevent-server
  CC       deflate.o
  CCLD     deflate
make[2]: Entering directory '/nghttp2-master/examples'
make[2]: Nothing to be done for 'install-exec-am'.
make[2]: Nothing to be done for 'install-data-am'.
make[2]: Leaving directory '/nghttp2-master/examples'
make[1]: Leaving directory '/nghttp2-master/examples'
Making install in python
make[1]: Entering directory '/nghttp2-master/python'
make[2]: Entering directory '/nghttp2-master/python'
make[2]: Nothing to be done for 'install-exec-am'.
make[2]: Nothing to be done for 'install-data-am'.
make[2]: Leaving directory '/nghttp2-master/python'
make[1]: Leaving directory '/nghttp2-master/python'
Making install in tests
make[1]: Entering directory '/nghttp2-master/tests'
Making install in testdata
make[2]: Entering directory '/nghttp2-master/tests/testdata'
make[3]: Entering directory '/nghttp2-master/tests/testdata'
make[3]: Nothing to be done for 'install-exec-am'.
make[3]: Nothing to be done for 'install-data-am'.
make[3]: Leaving directory '/nghttp2-master/tests/testdata'
make[2]: Leaving directory '/nghttp2-master/tests/testdata'
make[2]: Entering directory '/nghttp2-master/tests'
make[3]: Entering directory '/nghttp2-master/tests'
make[3]: Nothing to be done for 'install-exec-am'.
make[3]: Nothing to be done for 'install-data-am'.
make[3]: Leaving directory '/nghttp2-master/tests'
make[2]: Leaving directory '/nghttp2-master/tests'
make[1]: Leaving directory '/nghttp2-master/tests'
Making install in integration-tests
make[1]: Entering directory '/nghttp2-master/integration-tests'
make[2]: Entering directory '/nghttp2-master/integration-tests'
make[2]: Nothing to be done for 'install-exec-am'.
make[2]: Nothing to be done for 'install-data-am'.
make[2]: Leaving directory '/nghttp2-master/integration-tests'
make[1]: Leaving directory '/nghttp2-master/integration-tests'
Making install in doc
make[1]: Entering directory '/nghttp2-master/doc'
make[2]: Entering directory '/nghttp2-master/doc'
make[2]: Nothing to be done for 'install-exec-am'.
 /bin/mkdir -p '/usr/local/share/man/man1'
 /usr/bin/install -c -m 644 nghttp.1 nghttpd.1 nghttpx.1 h2load.1 '/usr/local/share/man/man1'
make[2]: Leaving directory '/nghttp2-master/doc'
make[1]: Leaving directory '/nghttp2-master/doc'
Making install in contrib
make[1]: Entering directory '/nghttp2-master/contrib'
rm -f nghttpx-init nghttpx-init.tmp
sed -e 's|@bindir[@]|/usr/local/bin|g' nghttpx-init.in > nghttpx-init.tmp
chmod +x nghttpx-init.tmp
mv nghttpx-init.tmp nghttpx-init
sed -e 's|@bindir[@]|/usr/local/bin|g' nghttpx.service.in > nghttpx.service
sed -e 's|@bindir[@]|/usr/local/bin|g' nghttpx-upstart.conf.in > nghttpx-upstart.conf
make[2]: Entering directory '/nghttp2-master/contrib'
make[2]: Nothing to be done for 'install-exec-am'.
make[2]: Nothing to be done for 'install-data-am'.
make[2]: Leaving directory '/nghttp2-master/contrib'
make[1]: Leaving directory '/nghttp2-master/contrib'
Making install in script
make[1]: Entering directory '/nghttp2-master/script'
make[2]: Entering directory '/nghttp2-master/script'
make[2]: Nothing to be done for 'install-exec-am'.
 /bin/mkdir -p '/usr/local/share/nghttp2'
 /usr/bin/install -c fetch-ocsp-response '/usr/local/share/nghttp2'
make[2]: Leaving directory '/nghttp2-master/script'
make[1]: Leaving directory '/nghttp2-master/script'
make[1]: Entering directory '/nghttp2-master'
make[2]: Entering directory '/nghttp2-master'
make[2]: Nothing to be done for 'install-exec-am'.
 /bin/mkdir -p '/usr/local/share/doc/nghttp2'
 /usr/bin/install -c -m 644 README.rst '/usr/local/share/doc/nghttp2'
make[2]: Leaving directory '/nghttp2-master'
make[1]: Leaving directory '/nghttp2-master'

```

**Comment by tatsuhiro-t** [comment]
You are right.  It is missing.  I thought that the following text indicates that but it isn't:

```
Libraries have been installed in:
   /usr/local/lib

If you ever happen to want to link against installed libraries
in a given directory, LIBDIR, you must either use libtool, and
specify the full pathname of the library, or use the `-LLIBDIR'
flag during linking and do at least one of the following:
   - add LIBDIR to the `LD_LIBRARY_PATH' environment variable
     during execution
   - add LIBDIR to the `LD_RUN_PATH' environment variable
     during linking
   - use the `-Wl,-rpath -Wl,LIBDIR' linker flag
   - have your system administrator add LIBDIR to `/etc/ld.so.conf'

See any operating system documentation about shared libraries for
more information, such as the ld(1) and ld.so(8) manual pages.
```

`make install` is generated task by automake/autoconf, so I'm not sure I can add extra command.
It could be operation system specific (what about *bsd?)
It would be good to mention this issue in README.

**Comment by Tapanito** [comment]
Should I create a PR with a patch to the README?

**Comment by tatsuhiro-t** [comment]
That would be awesome.

**Comment by tatsuhiro-t** [comment]
Addressed by #917
